### PR TITLE
dot/sync Request justification in block request

### LINKED ISSUE #1062: Request justification in block request message
## Expected Behavior

in `syncer.createBlockRequest` we should request the block justification.

note: currently, there is no way to verify the received justification without some sort of block number -> set ID mapping (will open another issue for that)

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Added request for justification in `syncer.createBlockRequest`
-
-

## Tests

```
go test ./dot/sync/...
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #1062

### CODE REVIEW

File: dot/sync/syncer.go [hunk 1]
```diff
@@ -270,7 +270,7 @@ func (s *Service) createBlockRequest(startInt int64) *network.BlockRequestMessag
 
 	blockRequest := &network.BlockRequestMessage{
 		ID:            randomID, // random
-		RequestedData: 3,        // block header + body
```
Comment (noot) [inline dot/sync/syncer.go:273]: could you declare `byte` consts for each of the possible data types and then `&` them together here?

File: dot/sync/message.go [hunk 2]
```diff
@@ -84,35 +84,35 @@ func (s *Service) CreateBlockResponse(blockRequest *network.BlockRequestMessage)
...
-		if (blockRequest.RequestedData & 1) == 1 {
+		if (blockRequest.RequestedData & network.RequestedDataHeader) == 1 {
 			retData, err := s.blockState.GetHeader(hash)
 			if err == nil && retData != nil {
 				blockData.Header = retData.AsOptional()
 			}
 		}
 		// body
-		if (blockRequest.RequestedData&2)>>1 == 1 {
+		if (blockRequest.RequestedData&network.RequestedDataBody)>>1 == 1 {
```
Comment (noot) [inline dot/sync/message.go:94]: could you add spaces around thhe `&` for consistency?

Comment (edwardmack) [reply dot/sync/message.go:94]: for some reason gofmt wants it this way, when I make lint with the spaces I get:
```
dot/sync/message.go:94: File is not `gofmt`-ed with `-s` (gofmt)
                if (blockRequest.RequestedData & network.RequestedDataBody)>>1 == 1 {
```

but when I try to make them all without a space I get:
```
dot/sync/message.go:87: File is not `gofmt`-ed with `-s` (gofmt)
                if (blockRequest.RequestedData&network.RequestedDataHeader) == 1 {
```
so this is what gofmt gives us, I don't know why.

Comment (noot) [reply dot/sync/message.go:94]: alright then lol the only diff is this has the bit shift 🤔 very strange, but ok as is then

**Pull request merged**